CompilerSource/parsing/parser.cpp

```diff
@@ -1922,32 +1892,22 @@ std::unique_ptr<AST::FunctionCallExpression> TryParseFunctionCallExpression(int
 }
 
 std::unique_ptr<AST::Node> TryParseControlExpression(SyntaxMode mode_) {
-  switch (mode_) {
-    case SyntaxMode::STRICT: {
-      require_token(TT_BEGINPARENTH, "Expected '(' before control expression");
-      auto expr = TryParseExpression(Precedence::kAll);
-      require_token(TT_ENDPARENTH, "Expected ')' after control expression");
-      return expr;
-    }
-    default:
-      herr->Error(token)
-          << "Internal error: unreachable (" __FILE__ ":" << __LINE__
-          << "): SyntaxMode " << (int) mode_ << " unknown to system";
-      [[fallthrough]];
-    case SyntaxMode::QUIRKS: {
-      auto operand = TryParseOperand();
-      if (map_contains(Precedence::kBinaryPrec, token.type) && token.type != TT_STAR) {
-        Token oper = token;
-        token = lexer->ReadToken(); // Consume the token
-        auto right = TryParseControlExpression(mode);
-        operand = std::make_unique<AST::BinaryExpression>(std::move(operand), std::move(right), oper.type);
-      }
-      // TODO: handle [] for array access and () for direct func call
-      return operand;
-    }
-    case SyntaxMode::GML:
-      return TryParseExpression(Precedence::kAll, nullptr);
+  if ((int)mode_ > 2) {
```

Comment:
Like below, this is another good place to use constants `(mode_ != STRICT && mode != QUIRKS)`. Otherwise, this looks like a very positive change. :slightly_smiling_face:

Reply:
Actually, there is another mode (`GML`); this is why I did that instead of a long condition.